### DISCUSSION TITLE
Applications crashing when DCPSInfoRepo dies

### DISCUSSION THREAD

**Discussion by jwilliams8067** [OpenDDS/OpenDDS#4290] [Q&A]
I am seeing an issue where my publishers and subscribers are seg faulting when the DCPSInfoRepo process dies.
I am using an older version of OpenDDS, specifically version 3.7 because of dependency limitations.
Was this a known issue, and has it been fixed ?

```gdb
backtrace of failure
Program terminated with signal 11, Segmentation fault.
#0  0x0021d8b7 in OpenDDS::DCPS::InfoRepoDiscovery::active() ()
    at InfoRepoDiscovery.cpp:369
369	InfoRepoDiscovery.cpp: No such file or directory.
	in InfoRepoDiscovery.cpp
(gdb) bt
#0  0x0021d8b7 in OpenDDS::DCPS::InfoRepoDiscovery::active() ()
    at InfoRepoDiscovery.cpp:369
#1  0x01317cad in OpenDDS::DCPS::Service_Participant::repository_lost(std::basic_string<char, std::char_traits<char>, std::allocator<char> >) ()
    at DCPS/Service_Participant.cpp:1030
#2  0x00219bee in OpenDDS::DCPS::FailoverListener::on_subscription_disconnected(DDS::DataReader*, OpenDDS::DCPS::SubscriptionLostStatus const&) ()
    at FailoverListener.cpp:120
#3  0x012d51c3 in OpenDDS::DCPS::DataReaderImpl::notify_subscription_disconnected(OpenDDS::DCPS::WriterIdSeq const&) () at DCPS/DataReaderImpl.cpp:2585
#4  0x01346b2d in OpenDDS::DCPS::DataLink::notify(OpenDDS::DCPS::DataLink::ConnectionNotice) () at DCPS/transport/framework/DataLink.cpp:794
#5  0x00f23b88 in OpenDDS::DCPS::TcpConnection::handle_close(int, unsigned long) () at TcpConnection.cpp:395
#6  0x00c305e6 in ACE_Select_Reactor_Handler_Repository::unbind (
    this=0x8ee5f5c, handle=17, pos=0x8ee688c, mask=1)
    at Select_Reactor_Base.cpp:392
#7  0x0131e1ad in ACE_Select_Reactor_Handler_Repository::unbind(int, unsigned long) () at /home/randyn/Desktop/ACE_wrappers/ace/Select_Reactor_Base.inl:40
#8  0x0132984b in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::remove_handler_i(int, unsigned long) ()
    at /home/randyn/Desktop/ACE_wrappers/ace/Select_Reactor_T.cpp:1019
#9  0x0132a8fc in ACE_Select_Reactor_T<ACE_Reactor_Token_T<ACE_Token> >::notify_---Type <return> to continue, or q <return> to quit---
```

**Comment by jrw972** [comment]
I haven't found a PR or a commit that addresses this issue.  However, this can only happen if you are using InfoRepo federation which has always had issues.

If possible, can you determine which pointer is null (`this` of InfoRepoDiscovery or the result of `get_dcps_info()`?  That might suggest a possible fix.

**Reply by jwilliams8067** [reply]
Yes I was attempting to setup 2 DCPSInfoRepo processes, but I only had one of them started at this time.
the this of InfoRepoDiscovery  does not appear to be null it is 0x8e90db0, I don't see get_dcps_info in the backtrace at all.

As a matter of fact, I have not been able to get the second DCPSInfoRepo to start, I consistently get errors about resolving the IOR,  Is there an alternative setup for failover ?

**Reply by jrw972** [reply]
I don't have any experience with Federation so the only thing I can offer you is https://opendds.readthedocs.io/en/latest-release/devguide/the_dcps_information_repository.html#repository-federation
